app/drivers/kscan/debounce.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <sys/util.h>
+
+#define DEBOUNCE_COUNTER_BITS 14
+#define DEBOUNCE_COUNTER_MAX BIT_MASK(DEBOUNCE_COUNTER_BITS)
```

Comment:
You mentioned not needing to document the max debounce period since 14 bits is so much, but AFAICT it'll still cause a pretty cryptic build error to exceed that (even though it wouldn't make sense to) so might want to either add back the documentation or change the check. Maybe taking the min of the config value and `2^14 - 1` could avoid the issue.

Reply:
Sure, it is a somewhat cryptic error (it just says the value is too large but doesn't say what the max is), but if you are trying to set a debounce time of over 16 seconds I have questions.